### PR TITLE
Fix rich editor focus handling

### LINKED ISSUE #2549: When editing existing strings in rich editor cursor jumps to the end
As reported on Matrix:

"When editing existing strings which have translations for "One" and "More", after each input, the cursor jumps to the end of the string."

The problem no longer appears after reverting back to f765fa994d71f6ea21c99dcd527ead4d0c4e1ea0.

### PR DESCRIPTION
**Pull request by eemeli**
Fixes #2549 by not "fixing" the cursor position during manual edits. Spent ages looking in entirely the wrong place for this.

Python changes are completely separate, and were required to account for errors discovered when running the server locally with no network connectivity.

### CODE REVIEW

**Comment by mathjazz** [comment]
> Python changes are completely separate, and were required to account for errors discovered when running the server locally with no network connectivity.

We need to code without internet access more often. :)

**Review by mathjazz — APPROVED** [review]
Nice work!

A comment or two would be nice, but not blocking.

File: translate/src/context/Editor.tsx [hunk 1]
```diff
@@ -350,10 +352,12 @@ function updateRichValue(
     }
 
     // Need to let react-dom "fix" the select position before setting it right
```
Comment (eemeli) [inline translate/src/context/Editor.tsx:354]: @mathjazz This comment is actually already in the code as an explainer. Now it's just more accurate.

Comment (mathjazz) [reply translate/src/context/Editor.tsx:354]: Yes, I meant on the backend.

**Pull request merged**